networking/firmware_protocol/src/lib.rs

```diff
@@ -81,41 +107,50 @@ impl SlimeString {
 pub struct Packet {
 	// TODO: This tag could really be dropped from the Rust side, but #[deku(temp)] is a bit wonky
 	tag: u32,
+	/// Sequence number for the packet. It is incremented for each subsequent packet and is used to reject out of order
+	/// packets. This is sometimes referred to as the packet id
 	seq: u64,
 	#[deku(ctx = "*tag")]
-	ty_: PacketType,
+	data: PacketData,
 }
 
 impl Packet {
-	pub fn new(seq: u64, ty_: PacketType) -> Packet {
+	pub fn new(seq: u64, data: PacketData) -> Packet {
 		Packet {
-			tag: ty_.deku_id().unwrap(),
+			tag: data.deku_id().unwrap(),
 			seq,
-			ty_,
+			data,
 		}
 	}
 
-	pub fn serialize_into(&self, buf: &mut [u8]) -> usize {
-		let bytes = self.to_bytes().unwrap();
+	/// Serialize the packet into a byte slice, returning the number of bytes written. If the packet cannot fit into
+	/// the buffer or data could not be serialied, Err is returned.
+	pub fn serialize_into(&self, buf: &mut [u8]) -> Result<usize, ()> {
+		// TODO: Deku should be extended to support in-place serialization instead of allocating here
+		let bytes = self.to_bytes().map_err(|_| ())?;
+		// Check we can fit the buffer
+		if bytes.len() > buf.len() {
+			return Err(());
+		}
 		buf[..bytes.len()].copy_from_slice(&bytes);
-		bytes.len()
+		Ok(bytes.len())
 	}
 
 	pub fn deserialize_from(buf: &[u8]) -> Result<Packet, ()> {
 		match Packet::from_bytes((buf, 0)) {
-			Ok(((tail, _), p)) if tail.len() == 0 => Ok(p),
+			Ok(((tail, _tail_offset), packet)) if tail.len() == 0 => Ok(packet),
 			_ => Err(()),
 		}
 	}
 
-	pub fn split(self) -> (u64, PacketType) {
-		(self.seq, self.ty_)
+	pub fn split(self) -> (u64, PacketData) {
+		(self.seq, self.data)
 	}
 }
 
 #[derive(Debug, PartialEq, DekuRead, DekuWrite)]
-#[deku(ctx = "e: deku::ctx::Endian, tag: u32", id = "tag", endian = "e")]
-pub enum PacketType {
+#[deku(ctx = "_: deku::ctx::Endian, tag: u32", id = "tag", endian = "big")]
```

Comment:
hmm, I assumed that hard coding the endian would mean that we can just completely remove this part of the context. If I'm wrong on that, the original way you had it is better.

Reply:
reverted.